cpp/src/arrow/builder.cc

```diff
@@ -815,48 +815,35 @@ Status BooleanBuilder::Append(const std::vector<bool>& values) {
 }
 
 // ----------------------------------------------------------------------
-// DictionaryBuilder
+// UniqueBuilder
 
 template <typename T>
-DictionaryBuilder<T>::DictionaryBuilder(const std::shared_ptr<DataType>& type,
-                                        MemoryPool* pool)
+UniqueBuilder<T>::UniqueBuilder(const std::shared_ptr<DataType>& type, MemoryPool* pool)
     : ArrayBuilder(type, pool),
```

Comment:
We may reach a point where it doesn't make sense to inherit from ArrayBuilder, and rather compose it (or not use it at all)